qiskit/circuit/quantumcircuit.py

```diff
@@ -687,6 +687,46 @@ def compose(self, other, qubits=None, clbits=None, front=False, inplace=False):
 
         return dest
 
+    def tensor(self, other):
+        """Tensor ``self`` with ``other``.
+
+        Args:
+            other (QuantumCircuit): The other circuit to tensor this circuit with.
+
+        Returns:
+            QuantumCircuit: the tensored circuit (returns None if inplace==True).
+        """
+        num_qubits = self.num_qubits + other.num_qubits
+        num_clbits = self.num_clbits + other.num_clbits
+
+        # If a user defined both circuits with via register sizes and not with named registers
+        # (e.g. QuantumCircuit(2, 2)) then we have a naming collision, as the registers are by
+        # default called "q" resp. "c". To still allow tensoring we define new registers of the
+        # correct sizes.
+        if len(self.qregs) == len(other.qregs) == 1 and \
+                self.qregs[0].name == other.qregs[0].name == 'q':
+            # check if classical registers are in the circuit
+            if num_clbits > 0:
+                dest = QuantumCircuit(num_qubits, num_clbits)
+            else:
+                dest = QuantumCircuit(num_qubits)
+
+        # handle case if ``measure_all`` was called on both circuits, in which case the
+        # registers are both named "meas"
+        elif len(self.cregs) == len(other.cregs) == 1 and \
+                self.cregs[0].name == other.cregs[0].name == 'meas':
+            cr = ClassicalRegister(self.num_clbits + other.num_clbits, 'meas')
+            dest = QuantumCircuit(*other.qregs, *self.qregs, cr)
+
+        # Now we don't have to handle any more cases arising from special implicit naming
+        else:
+            dest = QuantumCircuit(*other.qregs, *self.qregs, *other.cregs, *self.cregs)
+
+        # compose self onto the output, and then other
+        dest.compose(other, range(other.num_qubits), range(other.num_clbits), inplace=True)
+        return dest.compose(self, range(other.num_qubits, num_qubits),
+                            range(other.num_clbits, num_clbits))
```

Comment:
```suggestion
        dest.compose(self, range(other.num_qubits, num_qubits),
                            range(other.num_clbits, num_clbits), inplace=True)
        return dest
```
Can the second compose also be `inplace=True`? (It'd save an unneeded copy of `dest`.)